test/DualTest.jl

```diff
@@ -170,21 +170,28 @@ for N in (0,3), M in (0,4), V in (Int, Float32)
     @test one(NESTED_FDNUM) === Dual{TestTag()}(Dual{TestTag()}(one(PRIMAL), zero(M_PARTIALS)), zero(NESTED_PARTIALS))
     @test one(typeof(NESTED_FDNUM)) === Dual{TestTag()}(Dual{TestTag()}(one(V), zero(Partials{M,V})), zero(Partials{N,Dual{TestTag(),V,M}}))
 
-    @test rand(samerng(), FDNUM) === Dual{TestTag()}(rand(samerng(), V), zero(PARTIALS))
-    @test rand(samerng(), typeof(FDNUM)) === Dual{TestTag()}(rand(samerng(), V), zero(Partials{N,V}))
-    @test rand(samerng(), NESTED_FDNUM) === Dual{TestTag()}(Dual{TestTag()}(rand(samerng(), V), zero(M_PARTIALS)), zero(NESTED_PARTIALS))
-    @test rand(samerng(), typeof(NESTED_FDNUM)) === Dual{TestTag()}(Dual{TestTag()}(rand(samerng(), V), zero(Partials{M,V})), zero(Partials{N,Dual{TestTag(),V,M}}))
+    if V <: Integer
+        @test rand(samerng(), FDNUM) == rand(samerng(), value(FDNUM))
+        @test rand(samerng(), NESTED_FDNUM) == rand(samerng(), value(NESTED_FDNUM))
+    elseif V <: AbstractFloat
+        @test rand(samerng(), typeof(FDNUM)) === Dual{TestTag()}(rand(samerng(), V), zero(Partials{N,V}))
+        @test rand(samerng(), typeof(NESTED_FDNUM)) === Dual{TestTag()}(Dual{TestTag()}(rand(samerng(), V), zero(Partials{M,V})), zero(Partials{N,Dual{TestTag(),V,M}}))
+        @test randn(samerng(), typeof(FDNUM)) === Dual{TestTag()}(randn(samerng(), V), zero(Partials{N,V}))
+        @test randn(samerng(), typeof(NESTED_FDNUM)) === Dual{TestTag()}(Dual{TestTag()}(randn(samerng(), V), zero(Partials{M,V})),
+        zero(Partials{N,Dual{TestTag(),V,M}}))
+        @test randexp(samerng(), typeof(FDNUM)) === Dual{TestTag()}(randexp(samerng(), V), zero(Partials{N,V}))
+        @test randexp(samerng(), typeof(NESTED_FDNUM)) === Dual{TestTag()}(Dual{TestTag()}(randexp(samerng(), V), zero(Partials{M,V})),
+        zero(Partials{N,Dual{TestTag(),V,M}}))
+    end
 
     # Predicates #
     #------------#
 
     @test ForwardDiff.isconstant(zero(FDNUM))
-    @test ForwardDiff.isconstant(rand(FDNUM))
```

Comment:
For Integers this already tests that it's returning a real Array{Float64}, so this is unnecessary. For floating point numbers we've explicitly written in the test that it should have zero partials, so these tests were redundant on floats but failed on integers (due to no method for Array{Float64}), hence deleted.